Mathlib/Algebra/Algebra/Basic.lean

```diff
@@ -340,18 +396,18 @@ theorem algebraMap_smul (r : R) (m : M) : (algebraMap R A) r • m = r • m :=
 /-- If `M` is `A`-torsion free and `algebraMap R A` is injective, `M` is also `R`-torsion free. -/
 lemma NoZeroSMulDivisors.of_algebraMap_injective' {R A M : Type*} [CommSemiring R] [Semiring A]
     [Algebra R A] [AddCommMonoid M] [Module R M] [Module A M] [IsScalarTower R A M]
-    [NoZeroSMulDivisors A M] (h : Function.Injective (algebraMap R A)) :
+    [NoZeroSMulDivisors A M] [FaithfulSMul R A] :
     NoZeroSMulDivisors R M where
```

Comment:
This should be renamed, with the original name deprecated. Maybe R A M or at least A should be explicit?

Reply:
I realise now that this should not exist since it is just `NoZeroSMulDivisors.trans`. So I have dropped and added a deprecation (and we now have explicit vars).